src/GafferScene/SceneElementProcessor.cpp

```diff
@@ -96,6 +98,11 @@ void SceneElementProcessor::affects( const Plug *input, AffectedPlugsContainer &
 			outputs.push_back( it->get() );
 		}
 	}
+
+	if( input == outPlug()->childBoundsPlug() )
+	{
+		outputs.push_back( outPlug()->boundPlug() );
```

Comment:
Are we missing some here? `boundMethod` uses `filterValue`, but we're taking care of that above in a more generic fashion. `compute/hashBound` uses `childNames`, but I think that's taken care of by `SceneNode` (though did we way we like to be explicit in matching what we see in `compute*` anyway?). `hashBound` is using `objectPlug` in the `Union` case, and i couldn't see where that was factored in anywhere else up the class hierarchy (though I may have missed it). It also uses `boundPlug` in some cases.

Reply:
Good spot - in 7f9648a I've factored things out into our preferred conditional-per-output form and declared the additional dependencies.

Reply:
Much easier to follow, thanks :)